sql/tsql/TSqlParser.g4

```diff
@@ -3716,8 +3715,8 @@ table_source_item_joined
     ;
 
 table_source_item
-    : table_name_with_hint        as_table_alias?
-    | full_table_name             as_table_alias?
+    : full_table_name             deprecated_table_hint as_table_alias // this is currently allowed
+    | full_table_name             as_table_alias? ( with_table_hints? | deprecated_table_hint? | sybase_legacy_hints? )
```

Comment:
It can be simplified:

```suggestion
    | full_table_name             as_table_alias? (with_table_hints | deprecated_table_hint | sybase_legacy_hints)?
```